### ISSUE TITLE
Make logger extract more information from std::nested_exception

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#1011]
Currently this:

``` c++

static logging::logger logger("mylogger");

SEASTAR_TEST_CASE(test_exception_from_finally) {
    return make_ready_future<>().then([] {
        throw std::runtime_error("outer");
    }).finally([] {
        throw std::runtime_error("inner");
    }).then_wrapped([] (auto&& f) {
       try {
           f.get();
       } catch (...) {
           logger.error("failure: {}", std::current_exception());
       }
    });
}
```

Will print only:

```
ERROR 2016-03-09 15:05:06,699 [shard 0] mylogger - failure: std::nested_exception
```

Which is not useful at all.

**Comment by benoit-canet** [comment]
@tgrabiec In which file did you intented to put this unit test ?

**Comment by nyh** [comment]
@benoit-canet the unit test is just an example... The issue is that the code
```c++
logger.error("failure: {}", std::current_exception());
```
Doesn't print useful information when the exception is an std::nested_exception. In Tomek's example, he gave you an example program (in the form of a unit test) which generates such an exception.

**Comment by benoit-canet** [comment]
nyh: I understand, I wrote the patch for this and now I am looking forward to properly test it. So I wanted to know I there was an existing suitable place for this.

**Comment by nyh** [comment]
On Wed, Apr 13, 2016 at 5:14 PM, Benoît Canet <notifications@github.com>
wrote:

> nyh: I understand, I wrote the patch for this and now I am looking forward
> to properly test it. So I wanted to know I there was an existing suitable
> place for this.
>

This "test" doesn't check anything, it just prints out a message, so it's
not a very good unit test...
But if you really want to commit it (and not just use it during
development), probably the best place to put it is tests/futures_test.cc...

**Comment by slivne** [comment]
@tgrabiec , @benoit-canet  can we close this ?

**Comment by tgrabiec** [comment]
The patch was not merged, @benoit-canet was supposed to send v2.

**Comment by tgrabiec** [comment]
Fixed by ef8587a910d552b95d279f28974b679451e613b6
